File: packages/python-trace/api/logs.py
import requests
import json

# Log posting is only required for the Spolm hosted platform (tryspolm.com).
# For self-hosted deployments, pass base_url to Tracer() or set SPOLM_BASE_URL.
def post_log(api_key, agent_id, log, base_url: str = "https://api.tryspolm.com"):
    try:
        payload = {
            "agentId": agent_id,
            "logData": log,
        }

        headers = {
            "Authorization": f"Bearer {api_key}",
            "Content-Type": "application/json",
        }

        res = requests.post(
            f"{base_url}/api/logs/post",
            json=payload,
            headers=headers,
            timeout=10,
        )

        if res.ok:
            data = res.json()
            return {"valid": True, "data": data}

        try:
            error_body = res.json()
        except ValueError:
            error_body = res.text

        print("post_log failed response:", res.status_code, error_body)
        return {
            "valid": False,
            "message": (
                error_body.get("message")
                if isinstance(error_body, dict)
                else f"HTTP {res.status_code}"
            ),
        }

    except Exception as err:
        print("post_log error:", err)
        return {"valid": False, "message": str(err)}

File: packages/js-trace/api/logs.js
// Log posting is only required for the Spolm hosted platform (tryspolm.com).
// For self-hosted deployments, pass baseUrl to Tracer() or set SPOLM_BASE_URL.
async function postLog(apiKey, agentID, log, baseUrl = "https://api.tryspolm.com") {
  try {
    const payload = {
      agentId: agentID,
      logData: log,
    };
    const res = await fetch(`${baseUrl}/api/logs/post`, {
      method: "POST",
      headers: {
        "Authorization": `Bearer ${apiKey}`,
        "Content-Type": "application/json",
      },
      body: JSON.stringify(payload),
    });

    if (res.ok) {
      const data = await res.json();
      return { valid: true, data };
    }

    let errorBody = null;
    try {
      errorBody = await res.json();
    } catch (parseErr) {
      errorBody = await res.text().catch(() => null);
    }

    console.warn("postLog failed response:", res.status, errorBody);
    return { valid: false, message: errorBody?.message || `HTTP ${res.status}` };
  } catch (err) {
    console.error("postLog error:", err);
    return { valid: false, message: err.message };
  }
}

module.exports = { postLog };
